index.html

```diff
@@ -1380,7 +1380,11 @@ <h3>Types</h3>
 All <a>credentials</a>, <a>presentations</a>, and encapsulated objects MUST
 specify, or be associated with, additional more narrow <a>types</a> (like
 <code>UniversityDegreeCredential</code>, for example) so software systems can
-process this additional information.
+process this additional information. The specification of these additional more
+narrow types could be done in the object within which the additional information
+is found, rather than at a higher level such as the <code>type</code>
+<a>property</a> of the <a>credential</a> or <a>presentation</a>. It is also
+possible for the additional more narrow types to be implicit.
         </p>
 
         <p>
```

Comment:
Please add a definition of what an implicit type is in the context of a VC, and please provide an example of a VC with an implicit type in it.
Otherwise remove the concept of an implicit type as I do not know what it is or means.

Reply:
(Given yesterday's resolution, [the above comment](https://github.com/w3c/vc-data-model/pull/658#discussion_r292911089) should perhaps become a new issue, after application merging the PR in its current state.  My suggestion below will still apply.)

Perhaps --
`to be implicit.`
-> `to be implicit. Such implicit types may be implied by the Issuer by including, and inferred by the Verifier by evaluating the presence and/or value(s) of, properties of the implied type which are not described as part of the explicitly declared type(s).`

Reply:
From the above explanation, then I infer that every issuer can issue VCs of just the type VC, and from the properties in the VC, the verifier can infer what type of VC this is. Thus negating the need for the type property to ever be anything other than single valued.

Reply:
Yes.

Issuers *may* Issue VCs with any — or no — other declared (sub-)type.

Verifiers *may* likewise have policies which reject — or require! — VCs with — or lacking! — whatever (sub-)type declaration they like.

Verifiers *may* equally have policies which accept VCs of any — or no! — (sub-)type which were Issued by a particular Issuer and which contain whatever property/ies they like — including properties of a declared (sub-)type of which they know nothing except that property identifier.

----

As I've said previously in this thread and elsewhere, I have always understood the "or be associated with" phrasing to make this declaration (but not the association) optional.

You have said you have always understood otherwise -- that the declaration had to be somewhere nearby.

I submit that your interpretation makes the model less flexible, less usable, and indeed less likely to be widely adopted, and would prevent some kinds of use which I can foresee.

I submit that my interpretation does not prevent your interpretation from being used by anyone who concurs with it.

I believe that all of this leads to one conclusion — which basically accedes to yesterday's resolution, and accepts either yesterday's wording or my small addition above.

Reply:
Your proposed change is a significant normative change, because it invalidates the current text which states

All credentials, presentations, and encapsulated objects MUST specify, or be associated with, additional more narrow types (like UniversityDegreeCredential, for example) so software systems can process this additional information. 

You are now saying that "Issuers may Issue VCs with any — or no — other declared (sub-)type." which contradicts the current CR text.

If your proposed change is accepted this will mean a second CR is needed.

The only solution I can see to this dilemma is by creating a new subtype with the reserved word "Private" whose definition is something like "When the value of type property of the verifiable credential is Private, this means that the subtype has been determined privately between the issuer and the verifier and is not being specified in the type property"

Reply:
My text makes no normative change, and should not require a new CR.

> ... MUST specify, or be associated with, additional ...

-- is semantically equivalent to --

> ... MUST be associated with additional ...

-- as well as the somewhat more explicit, though apparently more confusing --

> ... MUST be associated, which association may be achieved by specific declaration or by inference, with additional ...

Your suggested requirement of a new "Private" subtype, on the other hand, _would be_ a normative change, and _would_ require a fresh CR.

Reply:
I think that @David-Chadwick's suggestion above should be a separate issue:
> Please add a definition of what an implicit type is in the context of a VC, and please provide an example of a VC with an implicit type in it.

And @TallTed's proposed additional text addressing it should be a separate PR.

We already have WG consensus recorded in a resolution accepting this PR as it currently stands.

Reply:
@brentzundel I fully agree with your original posting "My interpretation of the text". It was always intended that VCs and VPs had explicit typing. This was discussed at length and the WG should refer back to the original meeting minutes where this was discussed and agreed. (To remind you of the primary argument, it was so that a Verifier could quickly check the type to know if it could accept the VC or not, without having to process the entire VC first.) Implicit typing was never intended to be the semantics for the text "be associated with" and was as you said meant to refer to a higher type definition. The paragraph following the one containing "be associated with" makes it clear where this other type information should be found.

Changing explicit to implicit typing is a significant normative change to the spec and removes the benefit to the Verifier of fast processing. Therefore this PR should not be merged without other significant edits in this PR (or a companion one) that together ensure that a normative change is not made to the spec. Accepting this PR without also agreeing either further text in this PR or text in a companion PR will mean that two normative changes to the spec will have to be made in order to cancel each other out.